libuavcan/include/uavcan/node/generic_publisher.hpp

```diff
@@ -165,7 +168,9 @@ int GenericPublisher<DataSpec, DataStruct>::doEncode(const DataStruct& message,
 {
     BitStream bitstream(buffer);
     ScalarCodec codec(bitstream);
-    const int encode_res = DataStruct::encode(message, codec);
+    // if doing canfd transfer tail array optimisation is disabled
+    TailArrayOptimizationMode tao_mode = (!isStdOnly() && (getNode().isTaoDisabled() || getNode().isCanFdEnabled())) ? TailArrayOptDisabled:TailArrayOptEnabled;
```

Comment:
add a get_tao_mode() call

Reply:
this is the only place this is used, and information is gathered from different scopes.